mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -777,6 +787,20 @@ class LogisticRegressionModel private[spark] (
     throw new SparkException("No training summary available for this LogisticRegressionModel")
   }
 
+  @Since("2.1.0")
+  def binarySummary: BinaryLogisticRegressionTrainingSummary = summary match {
+    case b: BinaryLogisticRegressionTrainingSummary => b
+    case _ => throw new SparkException("Summary type dismatch, " +
+      "BinaryLogisticRegressionTrainingSummary expected.")
+  }
+
+  @Since("2.1.0")
+  def multinomialSummary: MultinomialLogisticRegressionTrainingSummary = summary match {
+    case m: MultinomialLogisticRegressionTrainingSummary => m
+    case _ => throw new SparkException("Summary type dismatch, " +
```

Comment:
We actually could return a valid multinomial summary, even for binary problems.  I'd recommend we do that.